### ISSUE TITLE
HeaderSearch has wrong styling when compiled

### ISSUE THREAD

**Issue by keehun** [carbon-design-system/carbon-components-svelte#945]
I am using a `HeaderSearch` component to include a search bar in my global header bar on top. When running svelte in local development, everything works properly and looks like this:

<img width="675" alt="Screen Shot 2021-12-23 at 2 24 53 PM" src="https://user-images.githubusercontent.com/77378/147288697-172d4aad-bcad-45e4-9383-55914ef66f28.png">

That is the right half of the browser window at the top of the page. Here it is expanded:
<img width="675" alt="Screen Shot 2021-12-23 at 2 25 34 PM" src="https://user-images.githubusercontent.com/77378/147288757-d766ff71-fe49-4f76-acbe-047c28f39081.png">

When I use `npm run build` and host it on Netlify, the `HeaderSearch` component is missing quite a few styles and looks like this:
<img width="679" alt="Screen Shot 2021-12-23 at 2 30 44 PM" src="https://user-images.githubusercontent.com/77378/147289119-e9a15aae-0005-401c-8bd9-7758dcbea4c0.png">

And here it is extended:
<img width="683" alt="Screen Shot 2021-12-23 at 2 31 08 PM" src="https://user-images.githubusercontent.com/77378/147289161-1f754bf8-81b5-45ed-a9c3-257ef2ae6b88.png">

I had run into a few, other minor differences between local & prod before, but nothing like this. Maybe something is not configured properly? I would love some pointers to figure out what's causing the differences here.

**Comment by metonym** [comment]
Are you using any Carbon preprocessors?

**Comment by metonym** [comment]
What type of set-up are you using? What value do you have for the `emitCSS` compiler options?

**Comment by keehun** [comment]
I'm using SvelteKit and `optimizeCss()` from `carbon-preprocess-svelte`. Using `netlify-adapter` for the SvelteKit adapter.

My `svelte.config.js` looks like this:

```js
import netlify from '@sveltejs/adapter-netlify';
import * as carbon from 'carbon-preprocess-svelte';

const production = process.env.NODE_ENV != 'development';

// Netlify will inject this for us, so only inject this for local builds.
if (!production) {
	process.env['BUILD_ID'] = 'LOCAL';
}

process.env['VITE_BUILD_ID'] = process.env['BUILD_ID'];

export default {
	preprocess: carbon.presetCarbon(),
	kit: {
		target: '#svelte',
		adapter: netlify(),
		vite: {
			optimizeDeps: { include: ['clipboard-copy'] },
			plugins: [production && carbon.optimizeCss()]
		}
	}
};
```

**Comment by metonym** [comment]
What happens if you remove the `optimizeCss` and re-build it?

**Comment by keehun** [comment]
That fixes it! Would this be a bug to file against `carbon-preprocess-svelte`? Or is it something wrong with `HeaderSearch` that's causing `optimizeCss` to optimize this out?

**Comment by metonym** [comment]
Yes – it seems that `optimizeCss` removes those incorrectly. Transferring this issue to carbon-preprocess-svelte.
